src/tsolvers/UFLATHandler.cc

```diff
@@ -31,3 +31,49 @@ lbool UFLATHandler::getPolaritySuggestion(PTRef pt) const {
     return l_Undef;
 }
 
+TRes UFLATHandler::check(bool full) {
+    auto res = TSolverHandler::check(full);
+    if (full and res == TRes::SAT) {
+        equalitiesToPropagate.clear();
+        ufsolver->collectEqualitiesFor(interfaceVars, equalitiesToPropagate, knownEqualities);
+        lasolver->collectEqualitiesFor(interfaceVars, equalitiesToPropagate, knownEqualities);
+    }
+    return res;
+}
+
+void UFLATHandler::declareAtom(PTRef atom) {
+    TSolverHandler::declareAtom(atom);
+    if (logic.isEquality(atom)) {
+        knownEqualities.insert(atom);
+    }
+}
+
+vec<PTRef> UFLATHandler::getSplitClauses() {
+    if (equalitiesToPropagate.size() == 0) {
+        return {};
+    }
+    vec<PTRef> res;
+    for (PTRef eq : equalitiesToPropagate) {
+        if (knownEqualities.count(eq) != 0) {
+            continue;
+        }
+        // create clauses corresponding to "x = y iff x >= y and x <= y"
+        assert(logic.isNumEq(eq));
+        PTRef lhs = logic.getPterm(eq)[0];
+        PTRef rhs = logic.getPterm(eq)[1];
+        assert((logic.isNumVar(lhs) or logic.isNumConst(lhs)) and (logic.isNumVar(rhs) or logic.isNumConst(rhs)));
+        PTRef leq = logic.mkLeq(lhs, rhs);
+        PTRef geq = logic.mkGeq(lhs, rhs);
+        vec<PTRef> args = {eq, logic.mkNot(leq), logic.mkNot(geq)}; // trichotomy clause
```

Comment:
I suspect this might cause a difference between GCC and Clang, I am not sure of the order if evaluation is fixed by the standard in branch initialization or not. To be sure, we should probably create the negation up front in a specified order.

Reply:
I have read a bit about this and it seems than unlike order of evaluation for function call arguments, the order of evaluation for brace-initialization is sequenced, so this should not be a problem.